File: app/src/main/java/meow/softer/mydiary/shared/gui/LetterComparator.java
package meow.softer.mydiary.shared.gui;

import meow.softer.mydiary.contacts.ContactsEntity;

import java.util.Comparator;

public class LetterComparator implements Comparator<ContactsEntity> {

    @Override
    public int compare(ContactsEntity lhs, ContactsEntity rhs) {
        return lhs.getSortLetters().compareTo(rhs.getSortLetters());

    }
    //comparator 用于比较两个对象
}

File: app/src/main/java/meow/softer/mydiary/shared/gui/LetterComparator.kt
package meow.softer.mydiary.shared.gui

import meow.softer.mydiary.contacts.ContactsEntity

class LetterComparator : Comparator<ContactsEntity> {
    override fun compare(lhs: ContactsEntity, rhs: ContactsEntity): Int {
        return lhs.sortLetters!!.compareTo(rhs.sortLetters!!)
    }
}
